### PR TITLE
Logging applied to more areas

### PR DESCRIPTION
**Pull request by azmigproject**

### CODE REVIEW

**Comment by prmathur-microsoft** [comment]
Please add more logging in - 
To figure out further we will need more logging in the following functions in AmqpsIotHubConnection and transport.
1.	sendMessage – please get as detailed as possible especially around send and encode. I believe the problem may be around this code.
2.	onLinkflow
3.	onDelivery – please clearly indicate the arrival of the message.
4.	invokeCallbacks from transport – indicate successful completion of this callback.
5.	sendMessageResult – please indicate the result (complete ?, error ?)
6.	handleMessage – indicate arrival of message and content
7.	messageSent – indicate success clearly and log failure indicating the reason.

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java [hunk 1]
```diff
@@ -354,15 +354,17 @@ public Integer sendMessage(Message message)
             byte[] tag = String.valueOf(this. nextTag++).getBytes();
             Delivery dlv = sender.delivery(tag);
```
Comment (prmathur-microsoft) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java]: Can we add logging on line 345 just before breaking that encoding is complete ? I suspect the issue is somewhere in this block.

**Comment by prmathur-microsoft** [comment]
Also, can you please sign the CLA for this request to get in.

**Comment by msftgits** [comment]
Hi, I am closing and re-opening this PR to bump the CLA bot. Sorry for the inconvenience!

**Pull request merged**